CMakeLists.txt

```diff
@@ -229,30 +225,84 @@ elseif (${HOST_OS} EQUAL ${OS_LINUX} OR ${HOST_OS} EQUAL ${OS_ANDROID})
     ) # todo: configure linux audio lib options
   
     if(NOT ANDROID)
-      list(APPEND osd_SRCS 
-		${d_core}/linux-dist/x11.cpp
-		${d_core}/linux-dist/main.cpp
-		${d_core}/linux-dist/evdev.cpp)
+      list(APPEND osd_SRCS ${d_core}/linux-dist/main.cpp)
 
-      add_definitions(-DSUPPORT_X11)  ## don't use GLES ?
-      link_libraries(X11)
+      link_libraries(${CMAKE_DL_LIBS})
 
-      find_package(PkgConfig)
-      pkg_check_modules(EGL egl)
-      if(EGL_FOUND)
+      find_package(OpenGL)
+
+      if(OpenGL_EGL_FOUND)
         add_definitions(-DSUPPORT_EGL)
+        list(APPEND osd_SRCS ${d_core}/utils/glinit/egl/egl.cpp)
+        link_libraries(OpenGL::EGL)
+      endif()
+
+      find_package(PkgConfig)
+
+      pkg_check_modules(ALSA alsa)
+      if(ALSA_FOUND)
+        add_definitions(-DUSE_ALSA)
+        list(APPEND osd_SRCS ${d_aout}/audiobackend_alsa.cpp)
+        include_directories(${ALSA_INCLUDE_DIRS})
+        link_libraries(${ALSA_LIBRARIES})
+      endif()
+
+      pkg_check_modules(AO ao)
+      if(AO_FOUND)
+        add_definitions(-DUSE_LIBAO)
+        list(APPEND osd_SRCS ${d_aout}/audiobackend_libao.cpp)
+        include_directories(${AO_INCLUDE_DIRS})
+        link_libraries(${AO_LIBRARIES})
+      endif()
+
+      pkg_check_modules(LIBEVDEV libevdev)
+      if(LIBEVDEV_FOUND)
+        add_definitions(-DUSE_EVDEV)
+        list(APPEND osd_SRCS ${d_core}/linux-dist/evdev.cpp)
+        include_directories(${LIBEVDEV_INCLUDE_DIRS})
+        link_libraries(${LIBEVDEV_LIBRARIES})
+      endif()
+
+      pkg_check_modules(LIBPULSE_SIMPLE libpulse-simple)
+      if(LIBPULSE_SIMPLE_FOUND)
+        add_definitions(-DUSE_PULSEAUDIO)
+        list(APPEND osd_SRCS ${d_aout}/audiobackend_pulseaudio.cpp)
+        include_directories(${LIBPULSE_SIMPLE_INCLUDE_DIRS})
+        link_libraries(${LIBPULSE_SIMPLE_LIBRARIES})
+      endif()
+
+      pkg_check_modules(LIBUDEV libudev)
+      if(LIBUDEV_FOUND)
+        add_definitions(-DUSE_UDEV)
+        include_directories(${LIBUDEV_INCLUDE_DIRS})
+        link_libraries(${LIBUDEV_LIBRARIES})
+      endif()
+
+      pkg_check_modules(X11 REQUIRED x11)
+      if(X11_FOUND)
+        add_definitions(-DSUPPORT_X11)  ## don't use GLES ?
+        list(APPEND osd_SRCS ${d_core}/linux-dist/x11.cpp)
+        include_directories(${X11_INCLUDE_DIRS})
+        link_libraries(${X11_LIBRARIES})
+        if(OpenGL_GLX_FOUND)
```

Comment:
I'd only fallback to GLX if EGL is not found. 

EGL can initialize both GLES and GL, GLX can only initialize GL.

EGL works over X11, afair, but it is a bit more fragile than GLX over older driver versions and such.

Reply:
What do you think about switching `SUPPORT_EGL` and `SUPPORT_GLX` in this case (and in the other places) to make EGL the default?
https://github.com/reicast/reicast-emulator/blob/d56d5bb9021205f2c25dc0f3e8b43b6f881186ff/libswirl/linux-dist/main.cpp#L356-L363

Reply:
Not sure if my question was clear since there is no feedback here. I suggest to write the lines above like this so EGL will be the default if supported:
```
#if defined(SUPPORT_EGL)
 	return egl_Init(x11_win, x11_disp); 
#elif defined(SUPPORT_GLX)
 	return glx_Init(x11_win, x11_disp); 
#else 
 	#error "only x11 supported" 
 	return true; 
 #endif
```

Reply:
`SUPPORT_GLX` also adds hooks in the x11 window creation - only one of them should be defined from the build system right now. See https://github.com/reicast/reicast-emulator/blob/d56d5bb9021205f2c25dc0f3e8b43b6f881186ff/libswirl/linux-dist/x11.cpp#L398-L413

Reply:
Done : GLX support added only if GLX is available and EGL is not.